Code/Framework/AzCore/Tests/EBus/TestEventProcessingPolicy.cpp

```diff
@@ -0,0 +1,173 @@
+        {
+            if (auto* reporter = AZ::Interface<ITestEventProcessingReporter>::Get(); reporter)
+            {
+                reporter->Report(iface->RTTI_GetTypeName());
+            }
+        }
+
+        template<class Results, class Function, class Interface, class... InputArgs>
+        static void CallResult(Results& results, Function&& func, Interface&& iface, InputArgs&&... args)
+        {
+            ReportIface(iface);
+            results = AZStd::invoke(AZStd::forward<Function>(func), AZStd::forward<Interface>(iface), AZStd::forward<InputArgs>(args)...);
+        }
+
+        template<class Function, class Interface, class... InputArgs>
+        static void Call(Function&& func, Interface&& iface, InputArgs&&... args)
+        {
+            ReportIface(iface);
+            AZStd::invoke(AZStd::forward<Function>(func), AZStd::forward<Interface>(iface), AZStd::forward<InputArgs>(args)...);
+        }
+    };
+
+    struct ITestHookEvents
+    {
+        AZ_RTTI(ITestHookEvents, "{349C00E8-65C7-4FF6-B505-E047238618CC}");
+
+        virtual void Contribute(AZStd::vector<AZStd::string>& stack) const = 0;
+    };
+
+    struct TestHookEventTraits
+        : public AZ::EBusTraits
+    {
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+        static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+        using EventProcessingPolicy = TestEventProcessingPolicy;
+    };
+
+    using TestHookEventBus = AZ::EBus<ITestHookEvents, TestHookEventTraits>;
+
+    struct TestOneEventHook
+        : public TestHookEventBus::Handler
+    {
+        AZ_RTTI(TestOneEventHook, "{BC6F4FCD-F035-4527-9262-7582ACE33EDF}", TestHookEventBus::Handler);
+
+        TestOneEventHook()
+        {
+            TestHookEventBus::Handler::BusConnect();
+        }
+
+        ~TestOneEventHook()
+        {
+            TestHookEventBus::Handler::BusDisconnect();
+        }
+
+        void Contribute(AZStd::vector<AZStd::string>& stack) const override
+        {
+            stack.push_back("TestOneEventHook");
+        }
+
+        const char* Type()
+        {
+            return "TestOneEventHook";
+        }
+    };
+
+    struct TestTwoEventHook
+        : public TestHookEventBus::Handler
+    {
+        AZ_RTTI(TestTwoEventHook, "{E6C52A54-56DA-4BB7-BEA5-B1CF07640A31}", TestHookEventBus::Handler);
+
+        TestTwoEventHook()
+        {
+            TestHookEventBus::Handler::BusConnect();
+        }
+
+        ~TestTwoEventHook()
+        {
+            TestHookEventBus::Handler::BusDisconnect();
+        }
+
+        void Contribute(AZStd::vector<AZStd::string>& stack) const override
+        {
+            stack.push_back("TestTwoEventHook");
+        }
+
+        const char* Type()
+        {
+            return "TestTwoEventHook";
+        }
+    };
+
+    struct TestReporter final
+        : public ITestEventProcessingReporter
+    {
+        TestReporter()
```

Comment:
This should be be using AZ_RTTI, since AZ::Interface needs it for the derived class?